File: python/treeTraversals.py
class TreeNode:
  def __init__(self, val = 0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right

  def inorder_traversal(root):
    if root:
      inorder_traversal(root.left)
      print(root.val)
      inorder_traveral(root.right)

  def preorder_traverasal(root):
    if root:
      print(root.val)
      preorder_traversal(root.left)
      preorder_traversal(root.right)

  def postorder_traversal(root):
    if root:
      postorder_traversal(root.left)
      postorder_traversal(root.right)
      print(root.val)

# example usage
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
root.left.left = TreeNode(4)
root.left.right = TreeNode(5)

print("In order traversal:")
inorder_traversal(root)

print("\n pre order traversal: ")
preorder_traversal(root)

print("\n post order traversal: ")
postorder_traversal(root)

File: javascript/treeTraversals.js
class Node
  {
    constructor(data)
    {
      this.data = data;
      this.left = null;
      this.right = null;
    }
  }
// in order traversal
function inorderTraversal(root)
{
  if(root == null)
  {
    return;
  }
  inorderTraversal(root.left);
  console.log(root.data);
  inorderTraversal(root.right);
}

// pre order traversal 
function preorderTraversal(root)
{
  if (root === null)
  {
    return;
  }
  console.log(root.data);
  preorderTraversal(root.left);
  preorderTraversal(root.right);
}

// post order traversal 
function postorderTraversal(root)
{
  if(root === null)
  {
    return; 
  }
  postorderTraversal(root.left);
  postorderTraversal(root.right);
  console.log(root.data);
}

// example usage 
const root = new Node(1);
root.left = new Node(2);
root.right = new Node(3);
root.left.left = new Node(4);
root.left.right = new Node(5);


inorderTraversal(root); // output; 4 2 5 1 3 
preorderTraversal(root); // output; 1 2 4 5 3 
postorderTraversal(root); // output; 4 5 2 3 1 
